core/include/drivers/gic.h

```diff
@@ -56,4 +56,74 @@ void gic_cpu_init(struct gic_data *gd);
 void gic_it_handle(struct gic_data *gd);
 
 void gic_dump_state(struct gic_data *gd);
+
+
+#if defined(CFG_ARM_GICV3)
+#if defined(ARM64)
+static inline uint32_t read_icc_ctlr_el1(void)
+{
+	uint32_t v;
+
+	__asm__ volatile ("mrs %0, S3_0_C12_C12_4" : "=r" (v));
+	return v;
+}
+
+static inline void write_icc_ctlr_el1(uint32_t v)
+{
+	__asm__ volatile ("msr S3_0_C12_C12_4, %0" : : "r" (v));
+}
+
+static inline void write_icc_pmr_el1(uint32_t v)
+{
+	__asm__ volatile ("msr S3_0_C4_C6_0, %0" : : "r" (v));
+}
+
+static inline uint32_t read_icc_iar0_el1(void)
+{
+	uint32_t v;
+
+	__asm__ volatile ("mrs %0, S3_0_c12_c8_0" : "=r" (v));
+	return v;
+}
+
+static inline void write_icc_eoir0_el1(uint32_t v)
+{
+	__asm__ volatile ("msr S3_0_c12_c8_1, %0" : : "r" (v));
+}
+#elif defined(ARM32)
+static inline uint32_t read_icc_ctlr_el1(void)
```

Comment:
These inline functions belongs in `core/arch/arm/include/arm32.h`.
You can guard them with `#if defined(CFG_ARM_GICV3)` if you like.
The names of the functions should be `read_<exact name of register>` or `write_<exact name of register>` to minimize confusion regarding which register is accessed. Abstraction of hardware is usually done elsewhere.